trace-dispatcher/doc/trace-dispatcher.md

```diff
@@ -108,689 +89,654 @@ data TraceAddBlockEvent blk =
+  severityFor   :: Namespace a -> Maybe a -> Maybe SeverityS
 
-`Info` is the default __severity__, in the absence of trace context or configured severity overrides.
+  privacyFor    :: Namespace a -> Maybe a -> Maybe Privacy
+  privacyFor _ _ =  Just Public
 
-NOTE: as en extension to the filtering severity type (`SeverityF`), a `Silence` constructor is defined, which encodes unconditional silencing of a particular trace -- and therefore serves as a semantic expression of the disabling tracers functionality of the old framework.
+  detailsFor    :: Namespace a -> Maybe a -> Maybe DetailLevel
+  detailsFor _ _ =  Just DNormal
 
-### Privacy
+  documentFor   :: Namespace a -> Maybe Text
 
-__Privacy__ annotations allows limiting __trace-outs__ that particular traces can reach.  It is expressed in terms of:
+  metricsDocFor :: Namespace a -> [(Text, Text)]
+  metricsDocFor _ = []
 
-```haskell
-data Privacy
-    = Confidential | Public
+  allNamespaces :: [Namespace a]
 ```
 
-__Confidential__ privacy level means that the trace will not be externalised from the system, except via __standard output__.
+## Metrics Integration
+
+Metrics are seamlessly incorporated into the system through regular trace messages implementing the `asMetrics` function within the `LogFormatting` typeclass. Unlike other trace components, metrics are not subjected to filtering and are consistently provided. This occurs as long as the `EKGBackend` is configured for the message. The `EKGBackend` then forwards these metrics to `cardano-tracer` for additional processing. Subsequently, they are dispatched as Prometheus metrics, extending their utility and visibility.
+
+## Frequency Limiting in Trace Filtering
+
+Frequency filtering is an integral aspect of trace filtering, offering an optional mechanism to limit the observable frequency of individual trace messages.
 
-The annotation mechanism is similar to the one of severity:
+In essence, this involves a fair and probabilistic suppression of messages within a particular trace when their moving-average frequency surpasses a specified threshold parameter.
+
+The frequency limiter, in addition to controlling message frequency, emits a suppression summary message under specific conditions:
+
+- When message suppression commences.
+- Every 10 seconds during active limiting, providing the count of suppressed messages.
+- When message suppression concludes, indicating the total number of suppressed messages.
+
+Each frequency limiter is assigned a name for identification purposes:
 
 ```haskell
-privately :: Trace m a -> Trace m a
-withPrivacy :: Monad m => (a -> Privacy) -> Trace m a -> Trace m a
+limitFrequency
+  :: forall a m . MonadUnliftIO m
+  => Double   -- messages per second
+  -> Text     -- name of this limiter
+  -> Trace m TraceDispatcherMessage -- the limiter's messages
+  -> Trace m a -- the trace subject to limitation
+  -> m (Trace m a) -- the original trace
```

Comment:
As a dev, am I supposed to utilize this in my tracer implementation? Or is this taken care of by configuration values and just provided as context here?